PowerEditor/src/Notepad_plus.cpp

```diff
@@ -1477,15 +1477,20 @@ void Notepad_plus::removeEmptyLine(bool isBlankContained)
 	_findReplaceDlg.processAll(ProcessReplaceAll, &env, isEntireDoc);
 
 	// remove the last line if it's an empty line.
-	if (isBlankContained)
+	auto lastLine = _pEditView->execute(SCI_GETLINECOUNT) - 1;
+	auto str2Search = isBlankContained ? TEXT("[\\r\\n]+^[\\t ]*$|^[\\t ]+$") : TEXT("[\\r\\n]+^$");
+	auto startPos = _pEditView->execute(SCI_POSITIONFROMLINE, lastLine - 1);
+	auto endPos = _pEditView->execute(SCI_POSITIONFROMLINE, lastLine) + _pEditView->execute(SCI_LINELENGTH, lastLine);

```

Comment:
Can't you use docLength?

Reply:
If you mean `_pEditView->execute(SCI_GETLENGTH)` then yes, would be simpler.

Reply:
Yes, I meant `auto endPos = _pEditView->execute(SCI_GETLENGTH)`.